src/plugins/intel_cpu/src/graph.cpp

```diff
@@ -940,6 +940,13 @@ void Graph::Allocate(const std::vector<size_t>& syncNodesInds) {
                 one_of(edge->getChild()->getType(), Type::Output, Type::MemoryOutput) &&
                 edge->inPlace(Edge::LOOK_DOWN)) {
                 edge->getChild()->resolveInPlaceEdges(Edge::LOOK_DOWN);
+            } else if (one_of(edge->getParent()->getType(), Type::MemoryInput)) {
+                auto memInp = std::dynamic_pointer_cast<node::MemoryInput>(edge->getParent());
+                if (memInp && memInp->haveSubgraph()) {
+                    // Since the ReadValueWithSubgraph is middle node, just add this branch in order to use
+                    // ProxyMemoryBlock to share memory
+                    edge->getParent()->resolveInPlaceEdges(Edge::LOOK_UP);
```

Comment:
Hi @maxnick ,
For example:
MemoryInput(ReadValueWithSubgraph)->computer nodes
MemoryInput(ReadValueWithSubgraph)->MemoryOutputStub
MemoryInput(ReadValueWithSubgraph)->Stateful
I have to add this branch to call [MemoryInput::resolveInPlaceEdges](https://github.com/xipingyan/openvino/blob/e1882764cb550bf12c3db5c323c4cfbe5e416bb6/src/plugins/intel_cpu/src/nodes/memory.cpp#L773), it will let `Stateful` and `MemoryOutputStub`, `computer nodes`, share memory. It is reasonable.  

If not call, these nodes's output will be inPlace, but MemoryOutputStub will 
 allocate memory, so maybe these output will empty ptr. Just clarify.

Reply:
The problem is in the `MemoryInput::selectOptimalPrimitiveDescriptor` implementation you developed for the subgraph scenario. There you:
1. Lose the inPlace port tag, when copy the output port config from the internal subgraph. Therefore the output edge inplace look up property is lost and the `resolveInPlaceEdges(Edge::LOOK_UP)` isn't being called for such a node.
2. It doesn't make sense to enforce the output port memory descriptor to mimic the internal subgraph output node memory descriptor, as the data are being read from the internal subgraph pretty rarely (only per reset state). So that it makes sense to stick with the external memory representation (just a default behavior of the the MemoryInput node), as the data will be read from the state via proxy memory manager more often.
Thus I would recommend the following:
1. Don't redefine selectPrimitiveDescriptor as it's more important to preserve the external graph memory representation to avoid extra reorders in runtime
2. Provide the MemoryInput node output memory descriptors to the subgraph output config, so that the subgraph does all the necessary memory layout alignment for us.

Reply:
1: Removed redefine `supportedPrimitiveDescriptors`
2: Internal subgraph provide input setting `InputConfig` interface. but there is no `OutputConfig `setting interface. 
@maxnick